frontend/src/helpers/user/user_api_wrappers.tsx

```diff
@@ -1,25 +1,295 @@
-/* -------------------------------------------------------------------------- */
-/*                      mock backend for user service                     */
-/* -------------------------------------------------------------------------- */
+import api from "@/helpers/endpoint";
+import { getLogger } from "@/helpers/logger";
+import { HTTP_METHODS, SERVICE } from "@/types/enums"; 
+import { revalidateTag } from "next/cache";
+import { Status, Role } from "@/types/enums";
+import User from "../../types/user";
+import HttpStatusCode from "@/types/HttpStatusCode";
+import { PeerPrepErrors } from "@/types/PeerPrepErrors";
+import { throwAndLogError } from "@/utils/errorUtils";
 
-const getProfileUrl = (username: string) => {
-  return "https://i.pravatar.cc/150?u=a042581f4e29026704d";
+const logger = getLogger("user_api_wrappers");
+
+const service = SERVICE.USER;
+const scope = [SERVICE.USER];
+
+const getUserByEmail = async (email: string, cache: RequestCache='no-cache'): Promise<User | undefined> => {
+  const response = await api({
+    method: HTTP_METHODS.GET,
+    service: service,
+    path: `email?email=${email}`,
+    tags: scope,
+    cache: cache
+  });
+
+  // successful response should return 200 with the user data
+  if (response.status === HttpStatusCode.OK) {
```

Comment:
Oh, as @c0j0s suggest [here](https://github.com/CS3219-AY2324S1/ay2324s1-course-assessment-g05/issues/47#issuecomment-1732220888), we can use `switch-case` to determine the type of error to throw, and we can shorten the code here. Do you want to refactor them?

Reply:
Ok! will do